### PR TITLE
Enable SECURITY_CONTEXT for OpenShift

### PR DESCRIPTION
**Pull request by kphalgun-px**
**What this PR does / why we need it**:
This PR sets SECURITY_CONTEXT to true for OpenShift, allowing Torpedo pods to run in privileged mode for write access to volume mounts, including hostPath volumes.

**Which issue(s) this PR fixes** (optional)
Closes #PTX-22444

**Special notes for your reviewer**:
Please review the following Jenkins build for the test case "SetupTearDown" in K8s using the link below:

https://jenkins.pwx.dev.purestorage.com/job/Users/job/Leela/job/tp-byoc/719/

Aetos Dashboard: https://aetos.pwx.purestorage.com/resultSet/testSetID/551426

Please review the following Jenkins build for the test case "UpgradeCluster" in OCP using the link below:

https://jenkins.pwx.dev.purestorage.com/job/Users/job/suhas/job/ocp/318/ (You will find no errors in creating log files in the /testresults volume mount)

Aetos Dashboard: https://aetos.pwx.purestorage.com/resultSet/testSetID/554870

### CODE REVIEW

**Review by mkoppal-px — APPROVED** [review]
LGTM

File: deployments/deploy-ssh.sh [hunk 1]
```diff
@@ -312,7 +312,7 @@ fi
 
 JUNIT_REPORT_PATH="/testresults/junit_basic.xml"
```
Comment (nikolaypopov) [inline deployments/deploy-ssh.sh:313]: I don't think for Openshift you can mount things to `/` dir, its gotta be something like `/tmp/` or `/mnt/`, please try this code on OCP jobs before merging

Comment (kphalgun-px) [reply deployments/deploy-ssh.sh:313]: Updated the description. Please review.

**Review by lsrinivas-pure — APPROVED** [review]
LGTM

**Review by nikolaypopov — APPROVED** [review]

**Pull request merged**